workflow-templates/appstore-build-publish.yml

```diff
@@ -68,24 +71,42 @@ jobs:
           php-version: ${{ env.PHP_VERSION }}
           coverage: none
 
+      - name: Install composer dependencies
+        # Continue if no composer.json
+        continue-on-error: true
+        run: |
+          cd ${{ env.APP_NAME }}
+          composer install --no-dev
+
       - name: Build ${{ env.APP_NAME }}
+        # Skip if no package.json
         if: ${{ steps.versions.outputs.nodeVersion }}
         run: |
           cd ${{ env.APP_NAME }}
           npm ci
           npm run build
 
       - name: Package ${{ env.APP_NAME }} ${{ env.APP_VERSION }}
+        # Try krankerl, fallback to makefile
         run: |
           cd ${{ env.APP_NAME }}
           krankerl package || make appstore
 
       - name: Checkout server ${{ fromJSON(steps.appinfo.outputs.result).nextcloud.min-version }}
+        continue-on-error: true
+        id: server-checkout
         run: |
           NCVERSION=${{ fromJSON(steps.appinfo.outputs.result).nextcloud.min-version }}
           wget --quiet https://download.nextcloud.com/server/releases/latest-$NCVERSION.zip
           unzip latest-$NCVERSION.zip
 
+      - name: Checkout server master fallback
+        uses: actions/checkout@v2
```

Comment:
Is this depth=0 / 1 by default?

Reply:
We're only checking out for occ command

Reply:
Yeah, but clone with depth=0 takes 10sec, without depth checks the whole history and takes like 2+ minutes atm

Reply:
Ah, sure!

Reply:
Docs says `fetch-depth: 0` fetches all commits, tags and history: https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches

Reply:
Yeah, default 1 seems to be the optimized one.